Path-Check/gaen-mobile
# Use tracingStrategy to access native module

#### Description:

We currently have a TracingStrategyContext that is responsible for
determining whether the tracing strategy is gps or GAEN via the
environment. This is no longer necessary now that this repo is
exclusively uses the GAEN strategy.

This commit:
- Removes the unnecessary TracingStrategyContext
- Updates strategy dependent contexts (like Exposure and Permissions
contexts) to import their strategy directly instead of relying on props.
- Updates direct references to the `nativeModule` to be called via the
tracingStrategy instead.


#### Linked issues:

Fixes: [use btnativemodule via tracing strategy](https://trello.com/c/lVJhV0aX/220-only-use-btnativemodule-via-tracingstrategy)


#### How to test:
- The app should build
- Tests should pass
- Affected user flow code input should work as expected
- Affected user flow submit diagnosis should work as expected
